src/test/java/uk/nhs/prm/e2etests/performance/RepositoryPerformanceTest.java

```diff
@@ -0,0 +1,124 @@
+        this.templatingService = templatingService;
+        this.mhsInboundQueue = mhsInboundQueue;
+        this.gp2gpMessengerOQ = gp2gpMessengerOQ;
+        this.ehrTransferServiceTransferCompleteOQ = ehrTransferServiceTransferCompleteOQ;
+        this.ehrTransferServiceUnhandledOQ = ehrTransferServiceUnhandledOQ;
+        this.ehrTransferServiceLargeEhrFragmentsOQ = ehrTransferServiceLargeEhrFragmentsOQ;
+        this.ehrTransferServiceSmallEhrOQ = ehrTransferServiceSmallEhrOQ;
+        this.ehrTransferServiceLargeEhrOQ = ehrTransferServiceLargeEhrOQ;
+        this.ehrTransferServiceNegativeAcknowledgementOQ = ehrTransferServiceNegativeAcknowledgementOQ;
+        this.ehrTransferServiceParsingDeadLetterQueue = ehrTransferServiceParsingDeadLetterQueue;
+    }
+
+    @BeforeAll
+    void init() {
+        ehrTransferServiceSmallEhrOQ.deleteAllMessages();
+        ehrTransferServiceLargeEhrOQ.deleteAllMessages();
+        ehrTransferServiceLargeEhrFragmentsOQ.deleteAllMessages();
+        ehrTransferServiceParsingDeadLetterQueue.deleteAllMessages();
+        ehrTransferServiceTransferCompleteOQ.deleteAllMessages();
+        ehrTransferServiceUnhandledOQ.deleteAllMessages();
+        ehrTransferServiceNegativeAcknowledgementOQ.deleteAllMessages();
+        gp2gpMessengerOQ.deleteAllMessages();
+    }
+
+    @Test
+    void Given_SuperLargeEhrWith100Fragments_When_PutIntoRepoAndPulledOut_Then_VisibleOnGp2gpMessengerOQ() {
+        // given
+        String patientNhsNumber = "9727018157";
+        EhrRequestTemplateContext ehrRequestTemplateContext = EhrRequestTemplateContext.builder()
+                .nhsNumber(patientNhsNumber)
+                .sendingOdsCode(TPP_PTL_INT.odsCode())
+                .asidCode(TPP_PTL_INT.asidCode()).build();
+        String ehrRequestMessage = this.templatingService.getTemplatedString(EHR_REQUEST, ehrRequestTemplateContext);
+        String outboundConversationId = ehrRequestTemplateContext.getOutboundConversationId();
+        log.info("OutboundConversationID: " + outboundConversationId);
+        ContinueRequestTemplateContext continueRequestTemplateContext =
+                ContinueRequestTemplateContext.builder()
+                        .outboundConversationId(outboundConversationId)
+                        .recipientOdsCode(Gp2GpSystem.REPO_DEV.odsCode())
+                        .senderOdsCode(TPP_PTL_INT.odsCode()).build();
+        String continueRequestMessage = this.templatingService.getTemplatedString(CONTINUE_REQUEST, continueRequestTemplateContext);
+
+        // when
+        this.repoService.addLargeEhrWithVariableManifestToRepo(patientNhsNumber, 100, TPP_PTL_INT.odsCode());
+        this.mhsInboundQueue.sendMessage(ehrRequestMessage, outboundConversationId);
+
+        sleepFor( 10000);
+
+        this.mhsInboundQueue.sendMessage(continueRequestMessage, outboundConversationId);
+
+        boolean messagesExist = this.gp2gpMessengerOQ.getAllMessagesFromQueueWithConversationIds(1, 100,
```

Comment:
Replace with `messagesFound` ? , because at the point that this code has finished running, the messages no longer exist on the queue (due to deleting the found messages).